exporter/factory.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"github.com/spf13/viper"
+
+	"github.com/census-instrumentation/opencensus-service/processor"
+)
+
+// TraceDataExporter inherits functions from TraceDataProcessor, and additionally
+// adds some exporter-specific functions. This helps the collector core to
+// identify which TraceDataProcessors are Exporters and which are internal
+// processing components, so that better validation of pipelines can be done.
+type TraceDataExporter interface {
+	processor.TraceDataProcessor
+
+	// GetExportFormat gets the name of the format in which this exporter sends its data.
+	GetExportFormat() string
+}
+
+// TraceDataExporterFactory is an interface that builds a new TraceDataExporter based on
+// some viper.Viper configuration.
+type TraceDataExporterFactory interface {
```

Comment:
Based on how golang works any factory that implements one of these interface will be able to be used as any other. So probably we need to change the name "NewFromViper" with something link "NewExporterFromViper" and equivalent for all others.

Reply:
I think since the return type is different, that is not the case here.